.dockerignore

```diff
@@ -1,15 +1,71 @@
+# Copy of .gitignore please keep these in sync
+*.css.tmp
+*.egg-info
+*.js.tmp
+*.less.css
+*.mo
+*.po~
+*.py[co]
+*.signed.zip
+*.styl.css
+*-all.css
+*-all.js
+*-min.css
+*-min.js
+.*.sw?
 .cache
+.DS_Store
 .env
-.git
 .ipython
-.ruff_cache
-.tox
+.mysql_history
+.nose*
+.npm/
+.pdbhistory
+.pytest_cache
+.ssh
+.tox/
+.vscode
+backups
+build*.py
 buildx-bake-metadata.json
 deps
-docker/artifacts
+docker*.yml
+docker/artifacts/*
 docs/_build
-logs
+docs/_gh-pages
+docs/api/_build
+local_settings.py
+MANIFEST
 node_modules
-site-static
-static-dist
+pip-log.txt
+private/
+.ruff_cache
+settings_local.py*
+settings_local_*.py
+shellng_local.py
+site-static/*
+src/olympia/discovery/strings.jinja2
+static-build/*
+static/css/node_lib/*
+static/js/node_lib/*
+storage/files/*
+storage/git-storage/*
+storage/guarded-addons/*
+storage/mlbf/*
+storage/shared_storage/*
+storage/sitemaps/*
+tmp/*
+
+# Additionally ignore these files from the docker build that are not in .gitignore
```

Comment:
Are these frequently updated?  (i.e. is there a real benefit to having these two lists of things to ignore, vs. just relying on .gitignore)

Reply:
When you say "these" you mean the files or the ignore file?

Dockerignore should include everything in gitignore because CI by definition only has what you committed.

Dockerignore should include other things not in gitignore because we don't necessarily want those files in the image. All of these fit into that category.

Reply:
I mean maintaining these two files with different lists in them - and specifically the differences between them, like these additional files.  Is there any benefit beyond not necessarily wanting them in the image?

Reply:
`.dockerignore` is a super set of `gitignore`. I thought about using a script to validate or auto generate the docker ignore but I think it's overkill.

The benefit is precisely excluding these files from the build. Excluding them from the build means fewer cache misses and faster builds and smaller images. I don't think we need an additional benefit as that is plenty.

Reply:
I added a dockerignore section to the docs. lmk wyt if this addresses any doubts or confusion.